src/object/cli_obj.c

```diff
@@ -1096,6 +1096,24 @@ obj_shards_2_fwtgts(struct dc_object *obj, uint32_t map_ver, uint8_t *bit_map,
 		req_tgts->ort_shard_tgts = req_tgts->ort_tgts_inline;
 	}
 
+	if (obj_auxi->spec_shard) {
+		D_ASSERT(grp_nr == 1);
+		D_ASSERT(shard_cnt == 1);
+		D_ASSERT(bit_map == NIL_BITMAP);
+		D_ASSERT(req_tgts->ort_srv_disp == 0);
+
+		tgt = req_tgts->ort_shard_tgts;
+
+		req_tgts->ort_grp_nr = 1;
+		req_tgts->ort_grp_size = 1;
+		if (obj_is_ec(obj))
+			req_tgts->ort_start_shard = rounddown(start_shard, obj_get_grp_size(obj));
```

Comment:
need to be "(start_shard / obj_get_grp_size(obj)) * daos_oclass_grp_size(oca);"?

Reply:
right